### PR TITLE
MULE-9168: Users should not be forced to use @Parameter to expose POJ…

### PR DESCRIPTION
**Pull request by juandesi**
…O fields

### CODE REVIEW

File: modules/extensions-support/src/test/java/org/mule/module/extension/internal/util/IntrospectionUtilsTestCase.java [hunk 1]
```diff
@@ -88,6 +99,29 @@ public void getNullFieldDataType() throws Exception
         IntrospectionUtils.getFieldDataType(null);
     }
 
+    @Test
+    public void getNoAnnotatedExposedPojoFields()
+    {
+        Collection<Field> exposedFields = IntrospectionUtils.getExposedFields(LifetimeInfo.class);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/module/extension/internal/util/IntrospectionUtilsTestCase.java]: you should always use hamcrest

File: modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java [hunk 2]
```diff
@@ -305,6 +309,27 @@ public static boolean isVoid(Method method)
...
+        if (!allFields.isEmpty())
+        {
+            return allFields;
+        }
+        try
+        {
+            PropertyDescriptor[] propertyDescriptors = Introspector.getBeanInfo(extensionType, Object.class).getPropertyDescriptors();
+            return Arrays.stream(propertyDescriptors)
+                    .filter(propertyDescriptor -> propertyDescriptor.getReadMethod() != null)
+                    .map(propertyDescriptor -> getField(extensionType, propertyDescriptor.getName(), propertyDescriptor.getPropertyType()))
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/module/extension/internal/util/IntrospectionUtils.java]: add a test case which verifies that field will never be null, even if the getter is not backed by a field of a consistent name

File: modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/SchemaBuilder.java [hunk 3]
```diff
@@ -775,14 +775,6 @@ protected void defaultOperation()
         };
     }
 
-    private boolean shouldGenerateChildElements(DataType dataType)
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/capability/xml/schema/SchemaBuilder.java]: buuug

File: modules/extensions-support/src/test/java/org/mule/module/extension/internal/util/IntrospectionUtilsTestCase.java [hunk 4]
```diff
@@ -88,6 +100,36 @@ public void getNullFieldDataType() throws Exception
...
+        assertThat(exposedFields, is(not(empty())));
+        assertThat(exposedFields.size(), is(2));
```
Comment (marianogonzalez) [inline modules/extensions-support/src/test/java/org/mule/module/extension/internal/util/IntrospectionUtilsTestCase.java]: not a reason to reject the PR, but just FYI, there's a hasSize matcher in the Matchers class
